File: src/main/java/com/cse3063f25grp1/context/Context.java
package com.cse3063f25grp1.context;

import java.util.List;
import java.util.Map;

import com.cse3063f25grp1.data.ChunkStore;
import com.cse3063f25grp1.model.Answer;
import com.cse3063f25grp1.model.Hit;
import com.cse3063f25grp1.model.Intent;
import com.cse3063f25grp1.model.Query;

public class Context {
    private Query question;
    private ChunkStore chunkStore;
    private Intent intent;
    private Map<Intent, List<String>> intentKeywordRules;
    private List<String> terms;
    private List<Hit> retrievedHits;
    private List<Hit> rerankedHits;
    private Answer finalAnswer;

    public Context() {
    }

    public Query getQuestion() {
        return question;
    }

    public void setQuestion(Query question) {
        this.question = question;
    }

    public ChunkStore getChunkStore() {
        return chunkStore;
    }

    public void setChunkStore(ChunkStore chunkStore) {
        this.chunkStore = chunkStore;
    }

    public Intent getIntent() {
        return intent;
    }

    public void setIntent(Intent intent) {
        this.intent = intent;
    }

    public Map<Intent, List<String>> getIntentKeywordRules() {
        return intentKeywordRules;
    }

    public void setIntentKeywordRules(Map<Intent, List<String>> intentKeywordRules) {
        this.intentKeywordRules = intentKeywordRules;
    }

    public List<String> getTerms() {
        return terms;
    }

    public void setTerms(List<String> terms) {
        this.terms = terms;
    }

    public List<Hit> getRetrievedHits() {
        return retrievedHits;
    }

    public void setRetrievedHits(List<Hit> retrievedHits) {
        this.retrievedHits = retrievedHits;
    }

    public List<Hit> getRerankedHits() {
        return rerankedHits;
    }

    public void setRerankedHits(List<Hit> rerankedHits) {
        this.rerankedHits = rerankedHits;
    }

    public Answer getFinalAnswer() {
        return finalAnswer;
    }

    public void setFinalAnswer(Answer finalAnswer) {
        this.finalAnswer = finalAnswer;
    }

}

File: iteration2/src/context/context.py
from typing import List, Dict
from src.model.query import Query
from src.model.intent import Intent
from src.model.hit import Hit
from src.model.answer import Answer
from src.data.chunk_store import ChunkStore

class Context:
    def __init__(self) -> None:
        self.__question: Query = None
        self.__chunk_store: ChunkStore = None
        self.__intent: Intent = None
        self.__intent_keyword_rules: Dict[Intent, List[str]] = None
        self.__terms: List[str] = None
        self.__retrieved_hits: List[Hit] = None
        self.__reranked_hits: List[Hit] = None
        self.__final_answer: Answer = None

    # Question
    def get_question(self) -> Query:
        return self.__question

    def set_question(self, value: Query) -> None:
        self.__question = value

    # ChunkStore
    def get_chunk_store(self) -> ChunkStore:
        return self.__chunk_store

    def set_chunk_store(self, value: ChunkStore) -> None:
        self.__chunk_store = value

    # Intent
    def get_intent(self) -> Intent:
        return self.__intent

    def set_intent(self, value: Intent) -> None:
        self.__intent = value

    # Intent keyword rules
    def get_intent_keyword_rules(self) -> Dict[Intent, List[str]]:
        return self.__intent_keyword_rules

    def set_intent_keyword_rules(self, value: Dict[Intent, List[str]]) -> None:
        self.__intent_keyword_rules = value

    # Terms
    def get_terms(self) -> List[str]:
        return self.__terms

    def set_terms(self, value: List[str]) -> None:
        self.__terms = value

    # Retrieved hits
    def get_retrieved_hits(self) -> List[Hit]:
        return self.__retrieved_hits

    def set_retrieved_hits(self, value: List[Hit]) -> None:
        self.__retrieved_hits = value

    # Reranked hits
    def get_reranked_hits(self) -> List[Hit]:
        return self.__reranked_hits

    def set_reranked_hits(self, value: List[Hit]) -> None:
        self.__reranked_hits = value

    # Final answer
    def get_final_answer(self) -> Answer:
        return self.__final_answer

    def set_final_answer(self, value: Answer) -> None:
        self.__final_answer = value
